src/group/class/discriminant.rs

```diff
@@ -0,0 +1,20 @@
+//! The fixed class group discriminant.
+use crate::num::mpz::Mpz;
+use std::str::FromStr;
+
+// 2048-bit prime, negated, congruent to 3 mod 4.  Generated using OpenSSL.
+// According to "A Survey of IQ Cryptography" (Buchmann & Hamdy) Table 1, IQ-MPQS for computing
+// discrete logarithms in class groups with a 2048-bit discriminant is comparable in complexity to
+// GNFS for factoring a 4096-bit integer.
```

Comment:
This is a pretty old paper (2001), and it's the single source that everyone cites for estimates of class group security. Tell me why I shouldn't be skeptical!

Reply:
Good point. The recent interest in class groups seems to have accelerated performance of algorithms for computing exponents (https://www.chia.net/2019/07/18/chia-vdf-competition-round-2-results-and-announcements.en.html). I don't see why that wouldn't also be the case for attacks, even independent of new algorithmic developments. I would be skeptical myself.

This paper presents a case against significant algorithmic improvements over IQ-MPQS for discrete log, but it's from 1999 and I haven't scrutinized it: https://www.iacr.org/archive/asiacrypt2003/07_Session07/05_149/28940064.pdf